reporting/impl/src/main/java/org/jboss/windup/reporting/rules/generation/techreport/TechReportService.java

```diff
@@ -218,11 +219,23 @@ public TechStatsMatrix getTechStatsMap(ProjectModel onlyForApplication)
             }
             else
             {
-                appsToCountTowards = Collections.singletonList(onlyForApplication.getId());
+                appsToCountTowards = Collections.singletonList(appBeingSummarized.getId());
             }
 
+            boolean isSharedApp = appsToCountTowards.size() > 1;
```

Comment:
Why here the concept of sharedApp doesnt have "shared" in the name of the app and only if the list of apps to be considered is greater than 1 ?

Reply:
There's a separate stat model in the list for a technology, for each app it applies to, and there's also a separate stat model if the library is shared ie, if there are 2 apps in an analysis sharing a library, there will be 3 stat models, and the shared library one applies to all 3 apps. Only the one which is for the shared stat applies to more than 1 app.

If it's the shared stat we only want the count to apply to the shared library app, not the actual analysed apps because their counts are covered by the other stat models for this technology. So we use a count of apps within the stat greater than 1 to flag that this is the shared library stat. When we know that, we only add the count from the stat to the appropriate places in the techstatsmap for the app with 'shared' in its name, not the other ones.

Reply:
+1